### ISSUE TITLE
Split At Headings But Keep Them

### ISSUE THREAD

**Issue by finn-matti** [lynchjames/note-refactor-obsidian#75]
The current way the reactoring is done breaks Links to headings. I think it would be better if the split at headings command would product the following contents. Suppose we have a note "Test1" like this:

```md
### Heading 1

FooBar

### Heading 2

ROFL
```

If I use "Split note by headings - H3" I would like the following result:

```md
### Heading 1

- [[Heading 1]]

### Heading 2

- [[Heading 2]]
```

In this way links like on other notes like [[Test1#Heading 1]] would continue to work even after a refactoring, which is imho more the spirit of refactoring anyways.

**Comment by sharik14** [comment]
I agree with @finn-matti 's comment. I'm just getting started comparing note-refactor and note-composer and have to say this would make all the difference in the world!

**Comment by AMGMNPLK** [comment]
+1
